modules/core/src/test/java/org/apache/ignite/logger/java/JavaLoggerFormatterTest.java

```diff
@@ -0,0 +1,178 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.logger.java;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import org.apache.ignite.testframework.junits.common.GridCommonTest;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Java log formatter test.
+ */
+@GridCommonTest(group = "Logger")
+public class JavaLoggerFormatterTest {
+
+    private static JavaLoggerFormatter formatter;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        formatter = new JavaLoggerFormatter();
+    }
+
+    /**
+     * Null category should fall back to anonymous
+     */
+    @Test
+    public void testNullCategory() {
+        assertFormattedOutput(null, JavaLoggerFormatter.ANONYMOUS_LOGGER_NAME);
+    }
+
+    /**
+     * Empty (zero-length) category should fall back to anonymous
+     */
+    @Test
+    public void testEmptyCategory() {
+        assertFormattedOutput("", JavaLoggerFormatter.ANONYMOUS_LOGGER_NAME);
+    }
+
+    /**
+     * Should use specified literal as category
+     */
+    @Test
+    public void testSimpleCategory() {
+        assertFormattedOutput("foo", "foo");
+    }
+
+    /**
+     * Should use classname as category
+     */
+    @Test
+    public void testClassCategory() {
+        assertFormattedOutput(JavaLoggerFormatterTest.class.getName(), "JavaLoggerFormatterTest");
+    }
+
+    /**
+     * Should append stack trace to log message
+     */
+    @Test
+    public void testThrowable() {
+        // given
+        final String msg = "Greed is good";
+        final String category = "probe";
+        final String exceptionMsg = "Not enough minerals";
+
+        LogRecord record = buildLogRecord(msg, category);
+        try {
+            throw new RuntimeException(exceptionMsg);
+        }
+        catch (Throwable t) {
+            record.setThrown(t);
+        }
+
+        // when
+        String result = formatter.format(record);
+
+        // then
+        final String expected = buildExpectedString(msg, category) + "\njava.lang.RuntimeException: " + exceptionMsg;
+        assertSimilar(expected, result);
+    }
+
+    /**
+     * Should use date format from property
+     */
+    @Test
+    public void testCorrectCustomFormatProperty() {
+        // given
+        System.setProperty("IGNITE_JAVA_LOGGER_DATE_FORMAT", "yyyy/MM/dd HH:mm:ss");
```

Comment:
There's a `WithSystemProperty` annotation, I think it should be used instead (but current approach is also fine)